src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -793,16 +791,46 @@ Status ServiceBasedObjectInfoAccessor::AsyncSubscribeToLocations(
   RAY_LOG(DEBUG) << "Subscribing object location, object id = " << object_id;
   RAY_CHECK(subscribe != nullptr)
       << "Failed to subscribe object location, object id = " << object_id;
-  auto status =
-      object_sub_executor_.AsyncSubscribe(subscribe_id_, object_id, subscribe, done);
+  auto on_subscribe = [subscribe](const std::string &id, const std::string &data) {
+    rpc::ObjectChange object_change;
+    object_change.ParseFromString(data);
+    std::vector<rpc::ObjectTableData> object_data_vector;
+    object_data_vector.emplace_back(object_change.data());
+    auto change_mode = object_change.is_add() ? rpc::GcsChangeMode::APPEND_OR_ADD
+                                              : rpc::GcsChangeMode::REMOVE;
+    gcs::ObjectChangeNotification notification(change_mode, object_data_vector);
+    subscribe(ObjectID::FromBinary(id), notification);
```

Comment:
better just change the original callback to use `rpc:ObjectChange`.

Reply:
Redis accessor also implements this interface. There is no rpc::objectchange there.